src/executables/util_service.c

```diff
@@ -1699,14 +1700,39 @@ process_server (int command_type, int argc, char **argv, bool show_usage, bool c
 	      else
 		{
 		  int pid;
-		  const char *args[] = { UTIL_CUBRID_NAME, token, NULL };
-		  status = proc_execute (UTIL_CUBRID_NAME, args, false, false, false, &pid);
 
-		  if (status == NO_ERROR && !is_server_running (CHECK_SERVER, token, pid))
+		  if (server_type_config::SINGLE_NODE ==
+		      (server_type_config) prm_get_integer_value (PRM_ID_SERVER_TYPE))
 		    {
-		      status = ER_GENERIC_ERROR;
+		      const char *args_page[] = { UTIL_CUBRID_NAME, "-t", "page", token, NULL };
```

Comment:
Add a comment explaining the purpose of the code.
```suggestion
		      // start both page server and transaction server
		      // page server must be started before transaction server
		      const char *args_page[] = { UTIL_CUBRID_NAME, "-t", "page", token, NULL };
```

The whole scope may be moved to a separate function (possible name `process_server_start_single_node ()`)

Reply:
i will move the code to a new function.